rust/crd/src/lib.rs

```diff
@@ -619,11 +635,12 @@ impl DruidCluster {
                 );
                 result.insert(MD_ST_HOST.to_string(), Some(mds.host.to_string()));
                 result.insert(MD_ST_PORT.to_string(), Some(mds.port.to_string()));
-                if let Some(user) = &mds.user {
-                    result.insert(MD_ST_USER.to_string(), Some(user.to_string()));
-                }
-                if let Some(password) = &mds.password {
-                    result.insert(MD_ST_PASSWORD.to_string(), Some(password.to_string()));
+                if mds.credentials_secret.is_some() {
+                    result.insert(MD_ST_USER.to_string(), Some(DB_USERNAME_PLACEHOLDER.into()));
+                    result.insert(
+                        MD_ST_PASSWORD.to_string(),
+                        Some(DB_PASSWORD_PLACEHOLDER.into()),
+                    );
```

Comment:
Was in there already, just complaining, not a fan of the `MD_ST_*` abbreviations... had to check what it meant.... :)

What happens if we did not provide user / password before or now do not provide the secret?
Does it work if we do not set `MD_ST_USER` and `MD_ST_PASSWORD` instead of setting an empty string?

Reply:
See f0b3e83 (I've renamed the constants and reduced their scope).

Reply:
I've only added the credentials where they used in tests (e.g. with postgrwsl): for` dbType=derby` no credentials are used/needed in the tests.